wit/consumer.wit

```diff
@@ -1,27 +1,14 @@
+/// The consumer interface allows a guest to dynamically update its subscriptions and configuration
 interface consumer {
-    // {client, message, channel, error, guest-configuration}
-    use messaging-types.{client, message, channel, error, guest-configuration};
-
-    /// Blocking receive for t-milliseconds with ephemeral subscription – if no message is received, returns None
-    subscribe-try-receive: func(c: client, ch: channel, t-milliseconds: u32) -> result<option<list<message>>, error>;
-
-    /// Blocking receive until message with ephemeral subscription
-    subscribe-receive: func(c: client, ch: channel) -> result<list<message>, error>;
-
-    /// 'Fit-all' type function for updating a guest's configuration – this could be useful for:
-    ///     - unsubscribing from a channel,
-    ///     - checkpointing,
-    ///     - etc..
-    update-guest-configuration: func(gc: guest-configuration) -> result<_, error>;
-
-    /// A message can exist under several statuses:
-    /// (1) available: the message is ready to be read,
-    /// (2) acquired: the message has been sent to a consumer (but still exists in the queue),
-    /// (3) accepted (result of complete-message): the message has been received and ACK-ed by a consumer and can be safely removed from the queue,
-    /// (4) rejected (result of abandon-message): the message has been received and NACK-ed by a consumer, at which point it can be:
-    ///         - deleted,
-    ///         - sent to a dead-letter queue, or
-    ///         - kept in the queue for further processing.
-    complete-message: func(m: message) -> result<_, error>;
-    abandon-message: func(m: message) -> result<_, error>;
+    use types.{error};
+
+    /// Set the current subscriptions for this guest.
+    /// 
+    /// Please note that implementations that provide `wasi:messaging` are responsible for ensuring
+    /// that guests are not allowed to subscribe to channels that they are not configured to
+    /// subscribe to (or have access to). Failure to do so can result in possible breakout or access
+    /// to resources that are not intended to be accessible to the guest. This means implementations
+    /// should validate that the configured topics are valid topics the guest should have access to or 
+    /// enforce it via the credentials used to connect to the service.
+    set-subscriptions: func(topics: list<string>) -> result<_, error>;
```

Comment:
```suggestion
    subscribe: func(topics: list<string>) -> result<_, error>;
```

Maybe? Calling it a "setter" feels weird

Reply:
+1 for `subscribe` over `set-subscriptions`.